iterreg/sparse/solvers.py

```diff
@@ -221,11 +227,14 @@ def ista(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000, f_store=1,
 # a priori jitting does not improve anything here
 # @njit
 def fista(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000, f_store=1,
-          verbose=False):
+          verbose=False, w_0=None):
     """Accelerated proximal gradient descent for the Tikhonov problem."""
     p = X.shape[1]
     L = norm(X, ord=2) ** 2
-    w = np.zeros(p)
+    if w_0 is None:
+        w = np.zeros(p)
+    else:
+        w = w_0
```

Comment:
beware that if w0 is defined outside, any modification to w here will affect w0 to in a side effect. It's safer to use `w = w0.copy()` IMO